x-pack/plugins/alerting/server/saved_objects/migrations.ts

```diff
@@ -333,6 +342,64 @@ function restructureConnectorsThatSupportIncident(
   };
 }
 
+function removeNullsFromSecurityRules(
+  doc: SavedObjectUnsanitizedDoc<RawAlert>
+): SavedObjectUnsanitizedDoc<RawAlert> {
+  const {
+    attributes: { params },
+  } = doc;
+  return {
+    ...doc,
+    attributes: {
+      ...doc.attributes,
+      params: {
+        ...params,
+        buildingBlockType: params.buildingBlockType != null ? params.buildingBlockType : undefined,
+        note: params.note != null ? params.note : undefined,
+        index: params.index != null ? params.index : undefined,
+        language: params.language != null ? params.language : undefined,
+        license: params.license != null ? params.license : undefined,
+        outputIndex: params.outputIndex != null ? params.outputIndex : undefined,
+        savedId: params.savedId != null ? params.savedId : undefined,
+        timelineId: params.timelineId != null ? params.timelineId : undefined,
+        timelineTitle: params.timelineTitle != null ? params.timelineTitle : undefined,
+        meta: params.meta != null ? params.meta : undefined,
+        query: params.query != null ? params.query : undefined,
+        filters: params.filters != null ? params.filters : undefined,
+        riskScoreMapping: params.riskScoreMapping != null ? params.riskScoreMapping : [],
+        ruleNameOverride: params.ruleNameOverride != null ? params.ruleNameOverride : undefined,
+        severityMapping: params.severityMapping != null ? params.severityMapping : [],
+        threat: params.threat != null ? params.threat : [],
+        threshold:
+          params.threshold != null &&
+          typeof params.threshold === 'object' &&
+          !Array.isArray(params.threshold)
```

Comment:
nit: I believe there is a lodash utility `isPlainObject()` or such, you can use for this test.